src/types/typeobject.rs

```diff
@@ -169,6 +189,53 @@ impl<'py> PyTypeMethods<'py> for Bound<'py, PyType> {
         name
     }
 
+    /// Gets the name of the module defining the `PyType`.
+    fn module(&self) -> PyResult<String> {
+        #[cfg(any(Py_LIMITED_API, PyPy, not(Py_3_13)))]
+        let name = self.getattr(intern!(self.py(), "__module__"))?.extract();
+
+        #[cfg(not(any(Py_LIMITED_API, PyPy, not(Py_3_13))))]
+        let name = {
+            use crate::ffi_ptr_ext::FfiPtrExt;
+            let obj = unsafe {
+                ffi::PyType_GetModuleName(self.as_type_ptr()).assume_owned_or_err(self.py())?
+            };
+
+            obj.extract()
+        };
+
+        name
+    }
+
+    /// Gets the [fully qualified name](https://docs.python.org/3/glossary.html#term-qualified-name) of the `PyType`.
+    fn fully_qualified_name(&self) -> PyResult<String> {
+        #[cfg(any(Py_LIMITED_API, PyPy, not(Py_3_13)))]
+        let name = {
+            let module = self.getattr(intern!(self.py(), "__module__"))?;
+            let qualname = self.getattr(intern!(self.py(), "__qualname__"))?;
+
+            let module_str = module.extract::<&str>()?;
```

Comment:
Unfortunately we can't extract `&str` on the limited API before 3.10, so we can use a helper type instead:

```suggestion
            let module_str = module.extract::<PyBackedStr>()?;
```